client/state/query-client-ssr.ts

```diff
@@ -1,19 +1,34 @@
-import { QueryClient, QueryCache, dehydrate } from 'react-query';
+import { QueryClient, QueryCache, dehydrate, hashQueryKey, TQueryKey } from 'react-query';
 import { MAX_AGE, BASE_STALE_TIME } from 'calypso/state/initial-state';
 const sharedCache = new QueryCache();
 
+class QueryClientSSR extends QueryClient {
+	fetchedQueryKeys: { [ hash: string ]: boolean } = {};
```

Comment:
I haven't found a way to get the `queries` that were called for each `queryClient` instance, so I extended the class and added the functionality.
It can then be used to determine what queries to send to the client.

Reply:
It's possible to add `QueryObserver`s to the query client or to the query cache. Maybe that could work? Add an observer to the query client, run the queries, in the observer record queries that were run, and finally collect their keys and select a subset of the cache.

I'm not sure at all if `QueryObserver` actually allows this -- it's not even certain that I can "observe all queries on this client" -- but certainly worth a look.

Reply:
@jsnajdr thanks for the tip! 
I had a look into it but it doesn't seem to do what we need it to: [QueryObserver](https://react-query-v3.tanstack.com/reference/QueryObserver#_top) is used to follow the changes of a specific query. 
There's also a [QueriesObserver](https://react-query-v3.tanstack.com/reference/QueriesObserver#_top) which is the same but you can pass it multiple queries to follow.
Trying to not specify the query(s) doesn't work either.

Reply:
This is golden stuff! Thank you!

I pushed a [commit](https://github.com/Automattic/wp-calypso/pull/68187/commits/7d066b4560940ef705a267a8a71b5aabda0baee4) to implement it.